R/grade_this.R

```diff
@@ -0,0 +1,70 @@
+
+# #' @export
+# last_result <- function(
+#   env = parent.frame()
+# ) {
+#   if (!exists(".result", envir = env, inherits = TRUE)) {
+#     stop("Could not find `.result`. Be sure to only use `last_result()` inside `grade_this()`")
+#   }
+#   get(".result", envir = env, inherits = TRUE)
+# }
+
+#' @export
+grade_this <- function(expr) {
+  # sometimes expr is a quosure already, so squash all quosures to a single expression
+  express <- rlang::get_expr(rlang::enquo(expr))
+
+  function(checking_env) {
+    eval_gradethis_expr({
+      rlang::eval_tidy(
+        express,
+        env = checking_env
+      )
+    })
+  }
+}
+
+#' @export
+grade_this_code <- function(
+  correct = getOption("gradethis.code.correct", getOption("gradethis.pass", "Correct!")),
+  incorrect = getOption("gradethis.code.incorrect", getOption("gradethis.fail", "Incorrect"))
+) {
+
+  # MUST wrap calling function to be able to shim in `.correct`/`.incorrect`
+  function(checking_env) {
+    checking_env[[".__correct"]] <- correct
+    checking_env[[".__incorrect"]] <- incorrect
+
+    grade_this({
```

Comment:
Can we add an argument that controls whether or not to add code feedback to wrong answers (if a solution exists) and set it to TRUE by default, controllable with an argument ot `grade_this()` like so:
```
grade_this({
...
}, code_feedback = FALSE)
```